src/mime/media_type.cr

```diff
@@ -0,0 +1,320 @@
+require "uri"
+require "http"
+require "mime"
+
+module MIME
+  # A MediaType describes a MIME content type with optional parameters.
+  struct MediaType
+    property media_type : String
+    property params : Hash(String, String)
+
+    # Creates a new `MediaType` instance.
+    def initialize(@media_type, @params = {} of String => String)
+    end
+
+    def to_s(io : IO)
+      io << media_type
+
+      params.keys.sort.each do |key|
```

Comment:
Because the order is insignificant, it makes sense to sort the parameters for display. Thus identical parameter sets will always print in the same way, no matter the order in which they were constructed.

Reply:
I don't think it's worth the performance loss. You should do this in `inspect`.

Reply:
Having a different order in `inspect` and `to_s` doesn't seem like a good idea.
I'd prefer to leave it as is for now. Sorting on `to_s` is simply an implementation detail demanded by the way these params are stored. `Hash` is just not a very efficient data structure for this application. But we currently don't have any alternatives in the stdlib and I don't think it's worth introducing a custom implementation for this specific use case.

Reply:
Completely disagree.

Reply:
Could you be more specific? Yes, it's less performant than it could be. But performance can be improved by changing the internal data handling. That's not required to be implemented right now.

Reply:
`to_s` is not for humans, it's for machines. We shouldn't make any performance concessions for humans. It's just how these methods are specified. Sort them for `inspect`.

Reply:
Do you really think it's a good idea to have different sort oder for `to_s` and `inspect`?

Reply:
Yes.